Vindicta.Altis/GameMode/GameModeBase.sqf

```diff
@@ -384,6 +384,68 @@ CLASS("GameModeBase", "MessageReceiverEx")
 			CALLM2(gMessageLoopGameMode, "addProcessCategoryObject", "GameModeProcess", _thisObject);
 		};
 
+		// Start a periodic check which will restart message loops if needed
+		[{CALLM0(_this#0, "_checkMessageLoops")}, [_thisObject], 2] call CBA_fnc_waitAndExecute;
+
+	} ENDMETHOD;
+
+	METHOD("_checkMessageLoops") {
+		params [P_THISOBJECT];
+
+		private _recovery = false;
+		private _crashedMsgLoops = [];
+
+		{													// Check all message loops created
+			private _msgLoop = T_GETV(_x);
+			if (!IS_NULL_OBJECT(_msgLoop)) then {
+				if(CALLM0(_msgLoop, "isNotRunning")) then {	// If it is not running, something bad has happened
+					CALLM0(_msgLoop, "unlock");				// Unlock it by force, it cant unlock itself anyway
+					_crashedMsgLoops pushBack _msgLoop;
+					OOP_ERROR_0("");
+					OOP_ERROR_0("! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! !");
+					OOP_ERROR_1("! ! ! THREAD IS NOT RUNNING: %1", GETV(_msgLoop, "name"));
```

Comment:
This works okay? Thought you couldn't have the commas in there.

Reply:
Yes. I think preprocessor first resolves the inner macro or something like that. Anyway, it works allright: (from my rpt file)
```
20:59:32 "[o_CivilWarGameMode_c2_s0_n[1].fnc] ERROR: ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! !"
20:59:32 "[o_CivilWarGameMode_c2_s0_n[1].fnc] ERROR: ! ! ! THREAD IS NOT RUNNING: Group AI thread"
20:59:32 "[o_CivilWarGameMode_c2_s0_n[1].fnc] ERROR: ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! ! !"
```